src/test/rpc/TransactionHistory_test.cpp

```diff
@@ -0,0 +1,154 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/Env.h>
+#include <test/jtx/envconfig.h>
+#include <ripple/protocol/JsonFields.h>
+#include <boost/container/static_vector.hpp>
+#include <algorithm>
+
+namespace ripple {
+
+class TransactionHistory_test : public beast::unit_test::suite
+{
+    void
+    testBadInput()
+    {
+        testcase("Invalid request params");
+        using namespace test::jtx;
+        Env env {*this, envconfig(no_admin)};
+
+        {
+            //no params
+            auto const result = env.client()
+                .invoke("tx_history", {})[jss::result];
+            BEAST_EXPECT(result[jss::error] == "invalidParams");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+
+        {
+            // test at 1 greater than the allowed non-admin limit
+            Json::Value params {Json::objectValue};
+            params[jss::start] = 10001; //limited to <= 10000 for non admin
+            auto const result = env.client()
+                .invoke("tx_history", params)[jss::result];
+            BEAST_EXPECT(result[jss::error] == "noPermission");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+    }
+
+    void testRequest()
+    {
+        testcase("Basic request");
+        using namespace test::jtx;
+        Env env {*this};
+
+        // create enough transactions to provide some
+        // history...
+        size_t const numAccounts = 20;
```

Comment:
Huh.  Does this work on all our compilers?  I didn't think you could use a local `const` integer as a template argument.  I would have expected it to work with a `static` or `constexpr` decorator, but not a plain `const`.

I tried to see if I could find something about this in the standard, but N4660 section 8.20 is too dense for me to make heads or tails out of.  I'll go with the standard being headed in the direction of allowing this if all 3 compilers support this.